iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/pipe/runtime/PipeMetaSyncer.java

```diff
@@ -104,16 +117,39 @@ private synchronized void sync() {
       pipeAutoRestartRoundCounter.set(0);
     }
 
-    final TSStatus status = procedureManager.pipeMetaSync();
+    final TSStatus metaSyncStatus = procedureManager.pipeMetaSync();
 
     if (somePipesNeedRestarting
```

Comment:
The "SUCCESS_STATUS" may be compacted to a single "if", and the "!" may be an "else".

Reply:
Besides, if there are "stopped" pipe, these pipe may be automatically restarted. Yet, there will be no pipe to restart given that there are no pipes in the cluster.

Reply:
> Besides, if there are "stopped" pipe, these pipe may be automatically restarted. Yet, there will be no pipe to restart given that there are no pipes in the cluster.

When there are "stopped" pipes in the cluster, should the result of `PipeTaskCoordinator.hasAnyPipe()` be true?

Reply:
Actually if the "successfullyRestart" is true, at least one pipe has a high probablity to exist... Namely, an empty pipe task won't let the "sync" method to reach "Successfully restart". Also, deleting the judgment won't cost much, at most the extra sync.